extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/UndertowBuildStep.java

```diff
@@ -239,6 +240,23 @@ public void kubernetes(HttpConfig config, BuildProducer<KubernetesPortBuildItem>
         portProducer.produce(new KubernetesPortBuildItem(config.port, "http"));
     }
 
+    @BuildStep
+    public void initCors(HttpConfig config, BuildProducer<FilterBuildItem> filterBuildItemProducer,
```

Comment:
You inject a runtime config at build time (`HttpConfig` is marked with the runtime phase). That's not something that should be done.

The filter would need to be added at runtime if we want to support that (and I think it makes sense to support it as at least the authorized domains could change from staging to production).

@stuartwdouglas do you have a vision on how to support this sort of things? AFAICS, in Undertow, all the filters are defined at static init.

Reply:
Yes, I agree with this filter being loaded at runtime. Configuration may change and It should not require recompiling.

I will wait for @stuartwdouglas comments before making more changes.

Reply:
If you want to do this at runtime use a ServletExtension. An example is at: https://github.com/quarkusio/quarkus/blob/ffab9f800049a9b59cdb3d3e8310e9cc2a6a40b7/extensions/elytron-security/runtime/src/main/java/io/quarkus/elytron/security/runtime/SecurityTemplate.java#L285

Basically you have a template that actually adds the filter (based on config), and then a build step that produces a ServletExtensionBuildItem.